src/main/java/com/google/sps/data/DeliverySlotManager.java

```diff
@@ -81,17 +91,22 @@ private void addDeliverySlotRequestToDatastore(DeliverySlot deliverySlot) {
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     Query query = new Query("deliverySlotRequest").setFilter(new Query.FilterPredicate("uid", Query.FilterOperator.EQUAL, userId));
-    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
+    List<Entity> results = datastore.prepare(query.addSort(DeliverySlot.Property.START_TIME.label, SortDirection.ASCENDING))
+      .asList(FetchOptions.Builder.withDefaults());
     
     for (Entity deliverySlotEntity : results) {
       DeliverySlot deliverySlot = new DeliverySlot((Date)deliverySlotEntity.getProperty(DeliverySlot.Property.START_TIME.label),
       (Date)deliverySlotEntity.getProperty(DeliverySlot.Property.END_TIME.label),
-      (String)deliverySlotEntity.getProperty(DeliverySlot.Property.USER_ID.label));
-
-      deliverySlot.setStartPoint((double)deliverySlotEntity.getProperty(DeliverySlot.Property.START_LAT.label),
-        (double)deliverySlotEntity.getProperty(DeliverySlot.Property.START_LNG.label));
-      deliverySlot.setSlotId(KeyFactory.keyToString(deliverySlotEntity.getKey()));
-      deliverySlots.add(deliverySlot);
+      (String)deliverySlotEntity.getProperty(DeliverySlot.Property.USER_ID.label),
+      /** canBeInThePast = */ true);
+      if (isExpired(deliverySlot)) {
+        datastore.delete(deliverySlotEntity.getKey());
```

Comment:
As discussed before, do not perform mutations in a method called `get`.

If you want to write a cron job cleaning expired slots that should be its own dedicated handler/helper written for this purpose. Don't try to bundle unexpected behavior in a method designed to fetch data from the datastore.